src/packages/console/src/components/newPlanDialog.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+
+import { Button, Text } from '@kibalabs/ui-react';
+import Box from '@material-ui/core/Box';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import { CardElement, ElementsConsumer } from '@stripe/react-stripe-js';
+import { Stripe, StripeElements } from '@stripe/stripe-js';
+
+import { IPlan } from '../consoleConfig';
+
+const useStyles = makeStyles((theme) => ({
+  upgradeDialogLoadingBox: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignContent: 'center',
+    margin: theme.spacing(2),
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+}));
+
+export interface INewPlanDialogProps {
+  // isOpen: boolean;
+  isUpgradeDialogLoading: boolean;
+  newPlan: IPlan;
+  currentPlan: IPlan;
+  upgradeCardError: string;
+  upgradeDiscountCode: string;
```

Comment:
this state doesnt need to be passed it, it can be managed in the dialog

Reply:
let me know if this makes sense, if it doesnt i'll take over once all the modals are moved into their own files

Reply:
upgradeCardError, upgradeDiscountCode , these can be managed